python/pypto/jit/specializer.py

```
@@ -1307,12 +1352,58 @@ def specialize(self) -> str:
+        for ctx, new_func in self._fn_origin:
+            gen_func = gen_funcs.get(ctx.func_name)
+            if ctx.orig_file is None or gen_func is None:
+                continue
+            gen_stmts = _dfs_stmts(gen_func)
+            orig_stmts = _dfs_stmts(new_func)
+            # Reparsing the unparsed transform must preserve statement count and
+            # DFS order; if it somehow diverges, skip this function rather than
+            # emit a wrong mapping (its spans then fall back to generated coords).
+            if len(gen_stmts) != len(orig_stmts):
+                continue
+            for gen_stmt, orig_stmt in zip(gen_stmts, orig_stmts):
```

Ruff (0.15.14)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)